src/services/lineService.ts

```diff
@@ -3,21 +3,22 @@ import apolloClient from '../util/ApolloClient';
 import LineFactory from '../factories/lineFactory';
 import NotificationType from '../enums/notificationType';
 import NotificationStore from '../stores/notificationStore';
+import { ILine } from '../models';
 
 export default class LineService {
-    public static async getAllLines() {
+    public static async getAllLines(): Promise<ILine[] | null> {
         try {
             const { data }:any = await apolloClient.query({ query: getLinjas });
             return LineFactory.linjasToILines(data.allLinjas.nodes);
         } catch (err) {
             NotificationStore.addNotification(
                 { message: 'Linjan haku ei onnistunut.', type: NotificationType.ERROR },
             );
-            return err;
+            return null;
```

Comment:
My opinion is that is it better to return null instead of any typed error, since we already have handled the error and showed a notification. Otherwise, the caller don't know what to expect.